functions/lib/stripe/create_account.py

```diff
@@ -0,0 +1,103 @@
+#########################
+# STRIPE CREATE ACCOUNT #
+#########################
+
+import json
+import uuid
+from dataclasses import dataclass
+
+import stripe
+from firebase_admin import firestore
+from firebase_functions import https_fn, options
+from google.cloud import firestore
+from google.cloud.firestore import DocumentReference, Transaction
+from lib.constants import db
+from lib.logging import Logger
+from lib.stripe.commons import ERROR_URL
+
+REFRESH_URL = "http://localhost:3000/stripe/refreshAccountLink"
+
+@dataclass
+class CreateStandardStripeAccountRequest:
+  returnUrl: str
+  organiser: str
+  
+  def __post_init__(self):
+    if not isinstance(self.returnUrl, str):
+      raise ValueError("Return Url must be provided as a string.")
+    if not isinstance(self.organiser, str):
+      raise ValueError("Organiser Id must be provided as a string.")
+      
+
+@firestore.transactional
+def check_and_update_organiser_stripe_account(transaction: Transaction, logger: Logger, organiser_ref: DocumentReference, return_url: str, refresh_url: str):
+
+  # Check if organiser exists and attempt to get details
+  maybe_organiser = organiser_ref.get(transaction=transaction)
```

Comment:
Don't think documentation supports passing in transaction=transaction as a parameter in a ref.get()

Below is the docs for a transaction object and using its get method with a DocRef object
https://firebase.google.com/docs/reference/js/v8/firebase.firestore.Transaction#get

Reply:
i think we let them do the complete onboarding with stripe, saves us building a workflow ourselves. Alternatively if you think it provides value, we could make this a separate ticket? what do you want?

Reply:
sorry i replied to wrong thread.. i think you can use a transaction in a `ref.get()`? hnmmm

copied from sample code here: https://firebase.google.com/docs/firestore/manage-data/transactions